Sources/MongoSwift/MongoClient.swift

```diff
@@ -402,6 +405,124 @@ public class MongoClient {
         )
     }
 
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+    @available(macOS 10.15, *)
+    internal class CmdHandler: CommandEventHandler {
+        private let continuation: AsyncStream<CommandEvent>.Continuation
+        internal init(continuation: AsyncStream<CommandEvent>.Continuation) {
+            self.continuation = continuation
+        }
+
+        // Satisfies the protocol
+        internal func handleCommandEvent(_ event: CommandEvent) {
+            self.continuation.yield(event)
+        }
+
+        internal func finish() {
+            self.continuation.finish()
+        }
+    }
+
+    @available(macOS 10.15, *)
+    internal class SDAMHandler: SDAMEventHandler {
+        private let continuation: AsyncStream<SDAMEvent>.Continuation
+        internal init(continuation: AsyncStream<SDAMEvent>.Continuation) {
+            self.continuation = continuation
+        }
+
+        // Satisfies the protocol
+        internal func handleSDAMEvent(_ event: SDAMEvent) {
+            self.continuation.yield(event)
+        }
+
+        internal func finish() {
+            self.continuation.finish()
+        }
+    }
+
+    /**
+     * Provides an `AsyncSequence` API for consuming command monitoring events.
+     *
+     * Example: printing the command events out would be written as
+     *  ```
+     *  for try await event in client.commandEventStream() {
+     *      print(event)
+     *  }
+     *  ```
+     * If you are looping over the events in the stream, you may wish to do so in a dedicated `Task`.
+     * The stream will be ended automatically if the `Task` it is running in is cancelled.
+     * - Returns: A `CommandEventStream` that implements `AsyncSequence`.
+     * - Note: Only the most recent 100 events are stored in the stream.
```

Comment:
What's the reasoning behind the choice of 100 here? Would there be any benefit in allowing users to set a custom value?